### PR TITLE
[BPK-4396]: BpkDialog a11y fix

### PR DESCRIPTION
**Pull request by olliecurtis**
The following PR resolves an A11y issue whereby when a dialog was being displayed it had no `aria-label` defined to attach to content for screenreaders.

As part of this PR some notable changes have been made in order to split out the `BpkDialog` from the `BpkModal` internal library to make them more independent and the code managable for each component.

Remember to include the following changes:

- [x] `UNRELEASED.md`
- [x] `README.md`
- [x] Tests
- [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

### CODE REVIEW

File: packages/bpk-component-modal/src/BpkModal.js [hunk 1]
```diff
@@ -24,14 +24,14 @@ import { withScrim } from 'bpk-scrim-utils';
 import { Portal, cssModules } from 'bpk-react-utils';
 
 import STYLES from './BpkModal.scss';
-import BpkModalDialog, {
+import BpkModalInner, {
   propTypes as modalDialogPropTypes,
   defaultProps as modalDialogDefaultProps,
   type Props as ModalDialogProps,
-} from './BpkModalDialog';
+} from './BpkModalInner';
```
Comment (olliecurtis) [inline packages/bpk-component-modal/src/BpkModal.js:31]: Renamed `BpkModalDialog` to `BpkModalInner` now that we have separated dialog and modal out

File: packages/bpk-component-dialog/src/BpkDialogInner.scss [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+@import '~bpk-mixins/index';
+
+.bpk-dialog {
```
[the remark this answers is not in the document]
Comment (shaundon) [reply packages/bpk-component-dialog/src/BpkDialogInner.scss:21]: That makes sense yeah, I think broadly the class name should match the component name – so maybe `.bpk-dialog-inner`?

**Review by shaundon — COMMENTED** [review]
Gonna test it out in storybook now

**Comment by shaundon** [comment]
Have run it in Storybook and all seems good, as far as I can see. Happy to approve once the feedback from earlier is addressed.

File: UNRELEASED.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 > Place your changes below this line.
 
+**Breaking:**
+- bpk-component-dialog:
+  - Added new required `ariaLabel` prop to `BpkDialog`
+  - Migrated `BpkDialog` from using `BpkModal` dependency for cleaner maintenance and separated functionality.
+
```
[the remark this answers is not in the document]
Comment (olliecurtis) [reply UNRELEASED.md:9]: Yeah I did some looking into this:
- None of the removed props are actually being used by any consumers of this 😌 
- The theming that was removed actually doesn't impact this component as looking at it, was just an export of the modal ones which only targeted the `BpkLink` which was only available to modals 🤷 so never actually themed this component 😆

**Comment by shaundon** [comment]
Test comment

File: UNRELEASED.md [hunk 4]
```diff
@@ -2,6 +2,11 @@
...
 ## How to write a good changelog entry
 
 1. Add 'Breaking', 'Added' or 'Fixed' in bold depending on if the change will be major, minor or patch according to [semver](semver.org).
```
Comment (shaundon) [inline UNRELEASED.md:12]: test comment

Comment (olliecurtis) [reply UNRELEASED.md:12]: Responding test comment

**Review by igunsaya — APPROVED** [review]
From my perspective, this would fix the issue 👍  I think long term, it would be good to revisit on how this component is designed so it has a clear heading/title

**Review by shaundon — APPROVED** [review]

**Pull request merged**